lib/puma/runner.rb

```diff
@@ -116,8 +116,8 @@ def redirect_io
         end
 
         STDERR.reopen stderr, (append ? "a" : "w")
-        STDERR.sync = true
         STDERR.puts "=== puma startup: #{Time.now} ==="
+        STDERR.flush unless STDERR.sync
```

Comment:
Likewise here, we still want the buffer flushed, but we don't want to be modifying STDOUT/STDERR

Reply:
Sorry, I don't get it, I think I'm missing something I don't sync that `STDERR.sync` does nothing it is just a getter :man_shrugging:

Reply:
Yes, `flush` probably does what we want, but `#sync` is false by default, hence the change in behaviour if it is never set to true

```
$ ruby -e 'p STDOUT.sync'
false
```

Reply:
Oops, I read the code as `STDERR.flush if STDERR.sync`, sorry! I understand my comments are confusing now.